Error building `mozjs_sys` on windows

Running `./mach check` (Windows 10, Intel x86_64 CPU) results in:

```
error: failed to run custom build command for `mozjs_sys v0.68.2 (https://github.com/servo/mozjs?rev=de147f0cf52da9c84fa95edf09c5f9f84695088c#de147f0c)`

Caused by:
  process didn't exit successfully: `C:\Users\p2c2e\Desktop\code\servo\target\debug\build\mozjs_sys-38b0f3cc97c850df\build-script-build` (exit code: 
101)
  --- stdout
  cargo:outdir=C:\Users\p2c2e\Desktop\code\servo\target\debug\build\mozjs_sys-b31d306cd726057c\out\build
  [[ 'C:\Users\p2c2e\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/configure -ot 'C:\Users\p2c2e\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/configure.in ]] && touch 'C:\Users\p2c2e\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/configure || true
  [[ 'C:\Users\p2c2e\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/old-configure -ot 'C:\Users\p2c2e\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/old-configure.in ]] && touch 'C:\Users\p2c2e\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/old-configure || true
  ! [[ 'C:\Users\p2c2e\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/configure.in -ot 'C:\Users\p2c2e\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/configure ]] && touch 'C:\Users\p2c2e\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/configure || true
  ! [[ 'C:\Users\p2c2e\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/old-configure.in -ot 'C:\Users\p2c2e\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/old-configure ]] && touch 'C:\Users\p2c2e\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/old-configure || true
  if [[ 'C:\Users\p2c2e\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/configure -nt config.status ]] ; then \
    PYTHON="C:\Users\p2c2e\AppData\Local\Programs\Python\Python39\python.exe" MOZ_TOOLS="/" \
    CC="clang-cl.exe" CFLAGS="-MD" \
    CPP="" CPPFLAGS="-IC:/Users/p2c2e/.cargo/registry/src/github.com-1ecc6299db9ec823/encoding_c_mem-0.2.6/include " \
    CXX="clang-cl.exe" CXXFLAGS="-MD" \
    AS="" AR="" \
    STLPORT_LIBS="" \
    'C:\Users\p2c2e\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/configure --disable-jemalloc --disable-js-shell --disable-tests --disable-export-js --disable-shared-js --build-backends=RecursiveMake --target=x86_64-pc-mingw32 --host=x86_64-pc-mingw32 || (cat config.log && exit 1) ; \
  fi

  --- stderr
   - Cannot opencat: config.log: No such file or directory
  mozmake: *** [C:\Users\p2c2e\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\makefile.cargo:194: maybe-configure] Error 1
  thread 'main' panicked at 'assertion failed: result.success()', C:\Users\p2c2e\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\build.rs:172:5  
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

by running 

``` $env:RUST_BACKTRACE='1';$env:PYTHON3='C:\Users\Tawfik\AppData\Local\Programs\Python\Python39\python.exe'; ./mach.bat build -d```

i get 

```
error: failed to run custom build command for `mozjs_sys v0.68.2 (https://github.com/servo/mozjs?rev=de147f0cf52da9c84fa95edf09c5f9f84695088c#de147f0c)`

Caused by:
  process didn't exit successfully: `D:\dev\servo\target\debug\build\mozjs_sys-38b0f3cc97c850df\build-script-build` (exit code: 101)
  --- stdout
  cargo:outdir=D:\dev\servo\target\debug\build\mozjs_sys-b31d306cd726057c\out\build
  [[ 'C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/configure -ot 'C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/configure.in ]] && touch 'C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/configure || true
  [[ 'C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/old-configure -ot 'C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/old-configure.in ]] && touch 'C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/old-configure || true
  ! [[ 'C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/configure.in -ot 'C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/configure ]] && touch 'C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/configure || true
  ! [[ 'C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/old-configure.in -ot 'C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/old-configure ]] && touch 'C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/old-configure || true
  if [[ 'C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/configure -nt config.status ]] ; then \
    PYTHON="C:\Users\Tawfik\AppData\Local\Programs\Python\Python39\python.exe" MOZ_TOOLS="/" \
    CC="clang-cl.exe" CFLAGS="-MD" \
    CPP="" CPPFLAGS="-IC:/Users/Tawfik/.cargo/registry/src/github.com-1ecc6299db9ec823/encoding_c_mem-0.2.6/include " \
    CXX="clang-cl.exe" CXXFLAGS="-MD" \
    AS="" AR="" \
    STLPORT_LIBS="" \
    'C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs'/js/src/configure --disable-jemalloc --disable-js-shell --disable-tests --disable-export-js --disable-shared-js --build-backends=RecursiveMake --target=x86_64-pc-mingw32 --host=x86_64-pc-mingw32 || (cat config.log && exit 1) ; \
  fi
  Creating Python 3 environment
  created virtual environment CPython3.9.6.final.0-64 in 5195ms
    creator CPython3Windows(dest=D:\Dev\servo\target\debug\build\mozjs_sys-b31d306cd726057c\out\build\_virtualenvs\init_py3, clear=False, no_vcs_ignore=False, global=False)
    seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=C:\Users\Tawfik\AppData\Local\pypa\virtualenv)
      added seed packages: pip==20.3.1, setuptools==57.0.0, wheel==0.36.2
    activators BashActivator,BatchActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator
  Error processing command. Ignoring because optional. (optional:packages.txt:comm/build/virtualenv_packages.txt)
  Traceback (most recent call last):
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\virtualenv.py", line 774, in <module>
      manager.populate()
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\virtualenv.py", line 466, in populate
      handle_package(package)
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\virtualenv.py", line 381, in handle_package
      submanager.populate(sitecustomize=sitecustomize)
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\virtualenv.py", line 466, in populate
      handle_package(package)
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\virtualenv.py", line 398, in handle_package
      f.write("%s\n" % os.path.relpath(path, python_lib))
    File "C:\Users\Tawfik\AppData\Local\Programs\Python\Python39\lib\ntpath.py", line 703, in relpath
      raise ValueError("path is on mount %r, start on mount %r" % (
  ValueError: path is on mount 'C:', start on mount 'D:'
  DEBUG: python3: running with pid 17156
  DEBUG: python3: sys.executable: 'c:\\Users\\Tawfik\\AppData\\Local\\Programs\\Python\\Python39\\python.exe'
  DEBUG: python3: executable from configuration: 'C:\\Users\\Tawfik\\AppData\\Local\\Programs\\Python\\Python39\\python.exe'
  DEBUG: python3: found executable: 'C:/Users/Tawfik/AppData/Local/Programs/Python/Python39/python.exe'
  INFO: Creating Python 3 environment
  INFO: created virtual environment CPython3.9.6.final.0-64 in 5195ms
  INFO:   creator CPython3Windows(dest=D:\Dev\servo\target\debug\build\mozjs_sys-b31d306cd726057c\out\build\_virtualenvs\init_py3, clear=False, no_vcs_ignore=False, global=False)
  INFO:   seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=C:\Users\Tawfik\AppData\Local\pypa\virtualenv)
  INFO:     added seed packages: pip==20.3.1, setuptools==57.0.0, wheel==0.36.2
  INFO:   activators BashActivator,BatchActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator
  INFO: Error processing command. Ignoring because optional. (optional:packages.txt:comm/build/virtualenv_packages.txt)
  INFO: Traceback (most recent call last):
  INFO:   File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\virtualenv.py", line 774, in <module>
  INFO:     manager.populate()
  INFO:   File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\virtualenv.py", line 466, in populate
  INFO:     handle_package(package)
  INFO:   File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\virtualenv.py", line 381, in handle_package
  INFO:     submanager.populate(sitecustomize=sitecustomize)
  INFO:   File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\virtualenv.py", line 466, in populate
  INFO:     handle_package(package)
  INFO:   File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\virtualenv.py", line 398, in handle_package
  INFO:     f.write("%s\n" % os.path.relpath(path, python_lib))
  INFO:   File "C:\Users\Tawfik\AppData\Local\Programs\Python\Python39\lib\ntpath.py", line 703, in relpath
  INFO:     raise ValueError("path is on mount %r, start on mount %r" % (
  INFO: ValueError: path is on mount 'C:', start on mount 'D:'

  --- stderr
  Traceback (most recent call last):
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\configure.py", line 231, in <module>
      sys.exit(main(sys.argv))
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\configure.py", line 55, in main
      sandbox.run(os.path.join(os.path.dirname(__file__), "moz.configure"))
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\configure\__init__.py", line 502, in run
      self.include_file(path)
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\configure\__init__.py", line 493, in include_file
      exec_(code, self)
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\util.py", line 55, in exec_
      exec(object, globals, locals)
    File "C:/Users/Tawfik/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/de147f0/mozjs/moz.configure", line 7, in <module>
      include("build/moz.configure/init.configure")
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\configure\__init__.py", line 853, in include_impl
      self.include_file(what)
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\configure\__init__.py", line 493, in include_file
      exec_(code, self)
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\util.py", line 55, in exec_
      exec(object, globals, locals)
    File "C:/Users/Tawfik/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/de147f0/mozjs/build/moz.configure/init.configure", line 241, in <module>
      def virtualenv_python3(env_python, virtualenv_name, build_env, mozconfig, help):
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\configure\__init__.py", line 836, in decorator
      depends = DependsFunction(self, func, dependencies, when=when)
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\configure\__init__.py", line 137, in __init__
      sandbox._value_for(self)
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\configure\__init__.py", line 614, in _value_for
      return self._value_for_depends(obj)
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\util.py", line 1049, in method_call
      cache[args] = self.func(instance, *args)
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\configure\__init__.py", line 623, in _value_for_depends
      value = obj.result()
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\util.py", line 1049, in method_call
      cache[args] = self.func(instance, *args)
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\configure\__init__.py", line 162, in result
      return self._func(*resolved_args)
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\configure\__init__.py", line 1298, in wrapped
      return new_func(*args, **kwargs)
    File "C:/Users/Tawfik/.cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/de147f0/mozjs/build/moz.configure/init.configure", line 354, in virtualenv_python3
      manager.build(python)
    File "C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\mozjs\python\mozbuild\mozbuild\virtualenv.py", line 554, in build
      raise Exception("Error populating virtualenv.")
  Exception: Error populating virtualenv.
  mozmake: *** [C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\makefile.cargo:194: maybe-configure] Error 1
  thread 'main' panicked at 'assertion failed: result.success()', C:\Users\Tawfik\.cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\build.rs:172:5
  stack backtrace:
     0: std::panicking::begin_panic_handler
               at /rustc/3e99439f4dacc8ba0d2ca48d221694362d587927\/library\std\src\panicking.rs:515
     1: core::panicking::panic_fmt
               at /rustc/3e99439f4dacc8ba0d2ca48d221694362d587927\/library\core\src\panicking.rs:92
     2: core::panicking::panic
               at /rustc/3e99439f4dacc8ba0d2ca48d221694362d587927\/library\core\src\panicking.rs:50
     3: build_script_build::build_jsapi
               at .\build.rs:172
     4: build_script_build::main
               at .\build.rs:57
     5: core::ops::function::FnOnce::call_once<fn(),tuple<>>
               at /rustc/3e99439f4dacc8ba0d2ca48d221694362d587927\library\core\src\ops\function.rs:227
  note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
war
```

Still getting errors, now `<utility>` is missing in `AlreadyAddRefed.h`. Will try to build from scratch one more time I guess.
```
cargo:warning=63 warnings generated.
  exit code: 0
  AR_x86_64-pc-windows-msvc = None
  AR_x86_64_pc_windows_msvc = None
  HOST_AR = None
  AR = None
  running: "C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Community\\VC\\Tools\\MSVC\\14.16.27023\\bin\\HostX64\\x64\\lib.exe" "-out:D:\\projects\\servo\\servo\\target\\debug\\build\\mozjs_sys-a1e117f0dc9d5253\\out\\build\\glue\\libjsglue.a" "-nologo" "D:\\projects\\servo\\servo\\target\\debug\\build\\mozjs_sys-a1e117f0dc9d5253\\out\\build\\glue\\src/jsglue.o"
  exit code: 0
  cargo:rustc-link-lib=static=jsglue
  cargo:rustc-link-search=native=D:\projects\servo\servo\target\debug\build\mozjs_sys-a1e117f0dc9d5253\out\build\glue
  CXXSTDLIB_x86_64-pc-windows-msvc = None
  CXXSTDLIB_x86_64_pc_windows_msvc = None
  HOST_CXXSTDLIB = None
  CXXSTDLIB = None
  Generting bindings ["./src/jsglue.hpp", "--rust-target", "1.40", "--rustified-enum", ".*", "--no-derive-default", "--enable-cxx-namespaces", "--generate", "functions,types,vars", "--ignore-methods", "--size_t-is-usize", "--rustfmt-configuration-file", "rustfmt.toml", "--", "-I", "D:\\projects\\servo\\servo\\target\\debug\\build\\mozjs_sys-a1e117f0dc9d5253\\out\\build\\dist/include", "-I", "D:\\projects\\servo\\servo\\target\\debug\\build\\mozjs_sys-a1e117f0dc9d5253\\out\\build\\js/src", "-x", "c++", "-fms-compatibility", "-DRUST_BINDGEN", "-DSTATIC_JS_API", "-std=c++17", "-DWIN32", "-D_CRT_USE_BUILTIN_OFFSETOF", "-include", "D:\\projects\\servo\\servo\\target\\debug\\build\\mozjs_sys-a1e117f0dc9d5253\\out\\build\\js/src/js-confdefs.h"] clang version 11.0.1.

  --- stderr
  Reticulating splines...
  Finished reading 60 moz.build files in 0.94s
  Read 0 gyp files in parallel contributing 0.00s to total wall time
  Processed into 224 build config descriptors in 0.18s
  RecursiveMake backend executed in 0.69s
    161 total backend files; 161 created; 0 updated; 0 unchanged; 0 deleted; 6 -> 38 Makefile
  Total wall time: 1.85s; CPU time: 0.62s; Efficiency: 34%; Untracked: 0.04s
  D:/cargo/git/checkouts/mozjs-fa11ffc7d4f1cc2d/de147f0/mozjs/intl/icu/source/common/unistr.cpp(1979,13): warning: unused function 'uprv_UnicodeStringDummy' [-Wunused-function]
  static void uprv_UnicodeStringDummy(void) {
              ^
  1 warning generated.
  D:\projects\servo\servo\target\debug\build\mozjs_sys-a1e117f0dc9d5253\out\build\dist/include\mozilla/AlreadyAddRefed.h:12:10: fatal error: 'utility' file not found
  D:\projects\servo\servo\target\debug\build\mozjs_sys-a1e117f0dc9d5253\out\build\dist/include\mozilla/AlreadyAddRefed.h:12:10: fatal error: 'utility' file not found, err: true
  thread 'main' panicked at 'Should generate JSAPI bindings OK: ()', D:\cargo\git\checkouts\mozjs-fa11ffc7d4f1cc2d\de147f0\build.rs:325:10
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[Warning] Could not generate notification!
```

--- a/README.md
+++ b/README.md
@@ -36,7 +36,7 @@ curl https://sh.rustup.rs -sSf | sh -s -- --default-toolchain none
 ```
 
 See also [Other installation methods](
-https://github.com/rust-lang-nursery/rustup.rs/#other-installation-methods)
+https://rust-lang.github.io/rustup/installation/other.html)
 
 ### Other dependencies
 
@@ -197,7 +197,7 @@ The `/m` will set it system-wide for all future command windows.
 
 2. Install virtualenv.
 
- In a normal Windows Shell (cmd.exe or "Command Prompt" from the start menu), do:
+ In a normal Windows Shell (cmd), do:
  ```
 pip install virtualenv
 ```
@@ -208,7 +208,7 @@ pip install virtualenv
 - [gstreamer-1.0-msvc-x86_64-1.16.0.msi](https://gstreamer.freedesktop.org/data/pkg/windows/1.16.0/gstreamer-1.0-msvc-x86_64-1.16.0.msi)
 - [gstreamer-1.0-devel-msvc-x86_64-1.16.0.msi](https://gstreamer.freedesktop.org/data/pkg/windows/1.16.0/gstreamer-1.0-devel-msvc-x86_64-1.16.0.msi)
 
-Note that the MinGW binaries will not work, so make sure that you install the MSVC the ones.
+Note that the MinGW binaries will not work, so make sure that you install the MSVC ones.
 
 Note that you should ensure that _all_ components are installed from gstreamer, as we require many of the optional libraries that are not installed by default.
 

--- a/README.md
+++ b/README.md
@@ -262,7 +262,7 @@ Run `./mach bootstrap-android --build` to get Android-specific tools. See wiki f
 [details](https://github.com/servo/servo/wiki/Building-for-Android).
 
 ### Cloning the Repo
-Your CARGO_HOME needs to point to (or be in) the same drive as your Servo repository (See #28530).
+Your CARGO_HOME needs to point to (or be in) the same drive as your Servo repository (See [#28530](https://github.com/servo/servo/issues/28530)).
 ``` sh
 git clone https://github.com/servo/servo
 cd servo

--- a/README.md
+++ b/README.md
@@ -261,38 +261,48 @@ linker = "lld-link.exe"
 Run `./mach bootstrap-android --build` to get Android-specific tools. See wiki for
 [details](https://github.com/servo/servo/wiki/Building-for-Android).
 
-## The Rust compiler
+### Cloning the Repo
+Your CARGO_HOME needs to point to (or be in) the same drive as your Servo repository (See #28530).
+``` sh
+git clone https://github.com/servo/servo
+cd servo
+```
+
+## Building
+
+Servo is built with [Cargo](https://crates.io/), the Rust package manager.
+We also use Mozilla's Mach tools to orchestrate the build and other tasks.
+You can call Mach like this:
+
+On Unix sytems:
+```
+./mach [command] [arguments]
+```
+On Windows Commandline:
+```
+mach.bat [command] [arguments]
+```
+The examples below will use Unix, but the same applies to Windows.
+
+### The Rust compiler
 
 Servo's build system uses rustup.rs to automatically download a Rust compiler.
 This is a specific version of Rust Nightly determined by the
 [`rust-toolchain`](https://github.com/servo/servo/blob/master/rust-toolchain) file.
 
-## Building
-
-Servo is built with [Cargo](https://crates.io/), the Rust package manager. We also use Mozilla's
-Mach tools to orchestrate the build and other tasks.
-
 ### Normal build
 
-To build Servo in development mode.  This is useful for development, but
-the resulting binary is very slow.
+To build Servo in development mode.
+This is useful for development, but the resulting binary is very slow:
 
 ``` sh
-git clone https://github.com/servo/servo
-cd servo
 ./mach build --dev
 ./mach run tests/html/about-mozilla.html
 ```
 
-Or on Windows MSVC, in a normal Command Prompt (cmd.exe):
-``` cmd
-git clone https://github.com/servo/servo
-cd servo
-mach.bat build --dev
-```
-
-For benchmarking, performance testing, or
-real-world use, add the `--release` flag to create an optimized build:
+### Release build
+For benchmarking, performance testing, or real-world use.
+Add the `--release` flag to create an optimized build:
 
 ``` sh
 ./mach build --release

--- a/README.md
+++ b/README.md
@@ -215,17 +215,13 @@ Note that you should ensure that _all_ components are installed from gstreamer, 
 4. Install Git for Windows (https://git-scm.com/download/win). DO allow it to add git.exe to the PATH (default
 settings for the installer are fine).
 
-5. Install Visual Studio Community 2017 (https://www.visualstudio.com/vs/community/). You MUST add "Visual C++" to the
-list of installed components as well as the "Windows Universal C runtime." They are not on by default. Visual Studio 2017 MUST installed to the default location or mach.bat will not find it.
-
-Note that version is hard to download online and is easier to install via [Chocolatey](https://chocolatey.org/install#installing-chocolatey) with:
+5. Install Visual Studio Build Tools 2019 (https://visualstudio.microsoft.com/de/downloads/#build-tools-for-visual-studio-2019). It is easiest to install via [Chocolatey](https://chocolatey.org/install#installing-chocolatey) with:
 ```
-choco install -y visualstudio2017community --package-parameters="'--add Microsoft.VisualStudio.Component.Git'"
-Update-SessionEnvironment #refreshing env due to Git install
-
-#--- UWP Workload and installing Windows Template Studio ---
-choco install -y visualstudio2017-workload-nativedesktop
+choco install -y visualstudio2019buildtools --package-parameters="--add Microsoft.VisualStudio.Component.Roslyn.Compiler --add Microsoft.Component.MSBuild --add Microsoft.VisualStudio.Component.CoreBuildTools --add Microsoft.VisualStudio.Workload.MSBuildTools --add Microsoft.VisualStudio.Component.Windows10SDK --add Microsoft.VisualStudio.Component.VC.CoreBuildTools --add Microsoft.VisualStudio.Component.VC.Tools.x86.x64 --add Microsoft.VisualStudio.Component.VC.Redist.14.Latest --add Microsoft.VisualStudio.Component.VC.ATL --add Microsoft.VisualStudio.Component.VC.ATLMFC --add Microsoft.VisualStudio.Component.TextTemplating --add Microsoft.VisualStudio.Component.VC.CoreIde --add Microsoft.VisualStudio.ComponentGroup.NativeDesktop.Core --add Microsoft.VisualStudio.Workload.VCTools"
 ```
+If you really need to use the Visual Studio Installer (UI), choose "Desktop development with C++" and add the optional "MSVC", "C++-ATL" and "C++-MFC" (latest).
+
+The Visual Studio 2019 Build Tools MUST be installed to the default location or mach.bat will not find them.
 
 ##### [Optional] Install LLVM for faster link times
 
@@ -242,13 +238,8 @@ linker = "lld-link.exe"
 
 ##### Troubleshooting a Windows environment
 
-> If you encountered errors with the environment above, do the following for a workaround:
-> 1.  Download and install [Build Tools for Visual Studio 2017](https://www.visualstudio.com/thank-you-downloading-visual-studio/?sku=BuildTools&rel=15)
-> 2.  Install `python3.9 x86-x64` and `virtualenv`
-> 3.  Run `mach.bat build -d`.
-
 >If you have troubles with `x64 type` prompt as `mach.bat` set by default:
-> 1. You may need to choose and launch the type manually, such as `x86_x64 Cross Tools Command Prompt for VS 2017` in the Windows menu.)
+> 1. You may need to choose and launch the type manually, such as `x86_x64 Cross Tools Command Prompt for VS 2019` in the Windows menu.)
 > 2. `cd to/the/path/servo`
 > 3. `python mach build -d`
 

--- a/python/servo/build_commands.py
+++ b/python/servo/build_commands.py
@@ -758,7 +758,7 @@ class MachCommands(CommandBase):
         return status
 
     @Command('clean',
-             description='Clean the build directory.',
+             description='Clean the target/ and python/_virtualenv[version]/ and support/hololens/ directories',
              category='build')
     @CommandArgument('--manifest-path',
                      default=None,
@@ -786,7 +786,7 @@ class MachCommands(CommandBase):
         return check_call(["cargo", "clean"] + opts, env=self.build_env(), verbose=verbose)
 
     @Command('clean-uwp',
-             description='Clean the support/hololens/ directory.',
+             description='Clean the support/hololens/ directory',
              category='build')
     def clean_uwp(self):
         uwp_artifacts = [

